datafusion/core/src/physical_plan/joins/hash_join.rs

```diff
@@ -362,45 +355,35 @@ impl ExecutionPlan for HashJoinExec {
         }
 
         let join_metrics = BuildProbeJoinMetrics::new(partition, &self.metrics);
-
-        // Initialization of operator-level reservation
-        self.reservation
-            .initialize("HashJoinExec", context.memory_pool());
-
-        // Inititalization of stream-level reservation
-        let reservation = SharedMemoryReservation::from(
-            MemoryConsumer::new(format!("HashJoinStream[{partition}]"))
-                .register(context.memory_pool()),
-        );
-
-        // Memory reservation for left-side data depends on PartitionMode:
-        // - operator-level for `CollectLeft` mode
-        // - stream-level for partitioned mode
-        //
-        // This approach allows to avoid cases when left data could potentially
-        // outlive its memory reservation and rely on `MemoryReservation` destructors
-        // for releasing memory in pool.
         let left_fut = match self.mode {
             PartitionMode::CollectLeft => self.left_fut.once(|| {
+                let reservation =
```

Comment:
I wonder if the same simplification can be applied to Nested Loops Joins and Symmetric Hash Join 🤔 

https://github.com/search?q=repo%3Aapache%2Farrow-datafusion%20SharedMemoryReservation&type=code